napari/utils/interactions.py

```diff
@@ -258,33 +269,39 @@ class Shortcut:
     instead of -.
     """
 
-    def __init__(self, shortcut: str):
+    def __init__(self, shortcut: KeyBindingLike):
         """
         Parameters
         ----------
-        shortcut : string
-            shortcut to format in the form of dash separated keys to press
+        shortcut : keybinding-like
+            shortcut to format
 
         """
-        self._values = re.split('-(?=.+)', shortcut)
-        for shortcut_key in self._values:
-            if (
-                len(shortcut_key) > 1
-                and shortcut_key not in KEY_SYMBOLS.keys()
-            ):
-
-                warnings.warn(
-                    trans._(
-                        "{shortcut_key} does not seem to be a valid shortcut Key.",
-                        shortcut_key=shortcut_key,
-                    ),
-                    UserWarning,
-                    stacklevel=2,
-                )
+        error_msg = trans._(
+            "{shortcut} does not seem to be a valid shortcut Key.",
+            shortcut=shortcut,
+        )
+        error = False
+
+        try:
+            self._kb = coerce_keybinding(shortcut)
+        except ValueError:
+            error = True
+        else:
+            for part in self._kb.parts:
+                shortcut_key = str(part.key)
+                if (
+                    len(shortcut_key) > 1
+                    and shortcut_key not in KEY_SYMBOLS.keys()
```

Comment:
Why `keys()` is used here?
```suggestion
                    and shortcut_key not in KEY_SYMBOLS
```

Reply:
idk, that code was written by someone else, i can remove it